File: JobsApp/Client/src/app/models/userParams.ts
import { Profession } from './../enums/profession.enum';
import { EmployerOrEmployee } from "../enums/employerOrEmployee.enum";
import { User } from "./User";

export class UserParams {
    employerOrEmployee: EmployerOrEmployee;
    minAge = 18;
    maxAge = 150;
    pageNumber = 1;
    pageSize = 5;
    profession: Profession;
    orderBy = 'lastActive';

    constructor ({employerOrEmployee , profession} : User) {
        this.employerOrEmployee = employerOrEmployee === EmployerOrEmployee.Employee ? EmployerOrEmployee.Employer : EmployerOrEmployee.Employee;
        this.profession = profession = Profession.Default;
    }
}

File: JobsApp/API/Helpers/UserParams.cs
namespace API.Helpers
{
    public class UserParams : PaginationParams
    {
         public string CurrentUsername { get; set; }
         
         public string EmployerOrEmployee { get; set; }

         public int MinAge { get; set; } = 18;
         public int MaxAge { get; set; } = 150;

         public string Profession { get; set; } = "Default";

        public string OrderBy { get; set; } = "lastActive";

        public string KnownAs { get; set; }
    }
}